application/cloud_api.py

```diff
@@ -68,9 +69,25 @@ def _writeIds(self):
     def uploadVideo(self,  video_path):
         print "uploadVideo called"
         with open(video_path, 'rb') as video:
-            files = {'video' : video}
-            r = requests.post(\
-                self.server_addr + 'uploadVideo', files = files, stream = True)
+            # We set the content-disposition 'filename' parameter manually
+            # incase we need to do streaming
+            files = {'video' : (os.path.basename(video_path), video)}
+
+            #Use a multipartencoder to stream the file data as just data
+            m = MultipartEncoder(fields = files)
+
+            # Get file size in MB's and check if it is greater than a 100 MB
+            # We use a 100MB size limit for transitioning to streaming rather
+            # than loading into memory
+            if m.len/(2**20) >= 100:
```

Comment:
I don't understand the math here on first glance. Comment what units m.len is.  Why 2^20?